src/__tests__/createActions-test.js

```diff
@@ -0,0 +1,212 @@
+import { createActions } from '../';
+import { isFSA } from 'flux-standard-action';
+
+describe('createActions', () => {
+  describe('createActions(actionMap)', () => {
+    it('returns an object with actions for each key', () => {
+      const a = createActions({
+        one: (b, c) => ({ bat: b, cat: c }),
+        two: {
+          payload: (d, e) => ({ dog: d, egg: e }),
+          meta: (/* d, e */) => ({ zed: 1 })
+        },
+        three: {
+          meta: (/* f */) => ({ zed: 2 })
+        }
+      });
+      expect(a).to.be.instanceOf(Object);
+      const a1 = a.one(1, 2);
+      expect(a1).to.eql({ type: 'one', payload: { bat: 1, cat: 2 } });
+      expect(isFSA(a1)).to.be.true;
+      expect(a.one.toString()).to.equal('one');
+      const a2 = a.two(3, 4);
+      expect(a2).to.eql({
+        type: 'two',
+        payload: { dog: 3, egg: 4 },
+        meta: { zed: 1 }
+      });
+      expect(isFSA(a2)).to.be.true;
+      expect(a.two.toString()).to.equal('two');
+      const a3 = a.three('foo');
+      expect(a3).to.eql({
+        type: 'three',
+        payload: 'foo',
+        meta: { zed: 2 }
+      });
+      expect(isFSA(a3)).to.be.true;
+      expect(a.three.toString()).to.equal('three');
```

Comment:
That's one of the great features of using redux-actions, you can reduce the
number of things you have to export in half, this one thing is both the
action creator and provides associated the constant type by calling
toString. You no longer have to export constants and action creators, one
serves for both purposes. It also reduces the risk that an action creator
is hooked up to the wrong action type since it is only typed once.

On Sunday, July 17, 2016, Victor Alvarez <notifications@github.com> wrote:

> In src/__tests__/createActions-test.js
> <https://github.com/acdlite/redux-actions/pull/104#discussion_r71076216>:
>
> > +      const a2 = a.two(3, 4);
> > +      expect(a2).to.eql({
> > +        type: 'two',
> > +        payload: { dog: 3, egg: 4 },
> > +        meta: { zed: 1 }
> > +      });
> > +      expect(isFSA(a2)).to.be.true;
> > +      expect(a.two.toString()).to.equal('two');
> > +      const a3 = a.three('foo');
> > +      expect(a3).to.eql({
> > +        type: 'three',
> > +        payload: 'foo',
> > +        meta: { zed: 2 }
> > +      });
> > +      expect(isFSA(a3)).to.be.true;
> > +      expect(a.three.toString()).to.equal('three');
>
> Why is testing toString() of the action creators important? As a library
> user I'd never use this value; I'd just care about the actions produced.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/acdlite/redux-actions/pull/104/files/628cf27017ee8dac86e5f1b08606b826e753e056#r71076216>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAWOQ8J4Dnq-MI-DQ31HJIFBubMm-etks5qWawWgaJpZM4JIZ-x>
> .
>


-- 
Jeff Barczewski
Founder of CodeWinds
http://codewinds.com/
Online developer training